Model/Api/GetProduct.php

```diff
@@ -0,0 +1,127 @@
+use Bolt\Boltpay\Helper\Hook as HookHelper;
+use Exception;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use \Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Webapi\Exception as WebapiException;
+use Magento\Framework\Webapi\Rest\Response;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+
+
+class GetProduct implements GetProductInterface
+{
+    /**
+     * @var GetProductDataInterface
+     */
+    private $productData;
+
+    /**
+     * @var ProductRepositoryInterface
+     */
+    protected $productRepositoryInterface;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var StockRegistryInterface
+     */
+    private $stockRegistry;
+
+    /**
+     * @var HookHelper
+     */
+    private $hookHelper;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @param ProductRepositoryInterface  $productRepositoryInterface
+     * @param StockRegistryInterface  $stockRegistry
+     * @param GetProductDataInterface  $productData
+     * @param StoreManagerInterface $storeManager
+     * @param HookHelper            $hookHelper
+     * @param Bugsnag               $bugsnag
+     */
+    public function __construct(
+        ProductRepositoryInterface  $productRepositoryInterface,
+        StockRegistryInterface  $stockRegistry,
+        GetProductDataInterface  $productData,
+        StoreManagerInterface $storeManager,
+        HookHelper $hookHelper,
+        Bugsnag $bugsnag
+    ) {
+        $this->productRepositoryInterface = $productRepositoryInterface;
+        $this->stockRegistry = $stockRegistry;
+        $this->productData = $productData;
+        $this->storeManager = $storeManager;
+        $this->hookHelper = $hookHelper;
+        $this->bugsnag = $bugsnag;
+    }
+
+    // TODO: ADD unit tests @ethan
+    /**
+     * Get user account associated with email
+     *
+     * @api
+     *
+     * @param string $productID
+     *
+     * @return \Bolt\Boltpay\Api\Data\GetProductDataInterface
+     *
+     * @throws NoSuchEntityException
+     * @throws WebapiException
+     */
+    public function execute($productID = '')
+    {
+        if (!$this->hookHelper->verifyRequest()) {
+            throw new WebapiException(__('Request is not authenticated.'), 0, WebapiException::HTTP_UNAUTHORIZED);
+        }
+
+        if ($productID === '') {
+            throw new WebapiException(__('Missing product ID in the request parameters.'), 0, WebapiException::HTTP_BAD_REQUEST);
+        }
+
+        try {
+            $storeId = $this->storeManager->getStore()->getId();
+            $product = $this->productRepositoryInterface->getById($productID, false, $storeId, false);
+            $this->productData->setProduct($product);
+            $stockItem = $this->stockRegistry->getStockItem($product->getId());
```

Comment:
I recall that it can be different ways to get inventory status for different M2 versions. @danac-gs  do you know about this?

Reply:
@vitaliyreznikov are you okay if we merge this so that @ethanwayda22 can begin end to end testing of the system that uses this endpoint?

Reply:
Yes, for sure

Reply:
@vitaliyreznikov @ethanwayda22 @njfaries 
`Magento\CatalogInventory\Api\StockRegistryInterface` and the whole `Magento_CatalogInventory`  got deprecated in 2.3 and replaced with [Multi Source Inventory](https://devdocs.magento.com/guides/v2.3/inventory/index.html) . Although there is some compatibility, this approach will not return correct stock information for the current store if the customer uses MSI. `\Magento\CatalogInventory\Model\Spi\StockRegistryProviderInterface::getStockStatus` should be used instead as it will provide the correct information as a summary due to `\Magento\InventoryCatalog\Plugin\CatalogInventory\Api\StockRegistry\AdaptGetStockStatusPlugin`

Reply:
Thank you @danac-gs, will update these calls in the V1 endpoint